### ISSUE TITLE
--fail-on any does not fail on warnings such as MD057 when using `check --fix`

### ISSUE THREAD

**Issue by rmuir** [rvben/rumdl#435]
### rumdl version

rumdl 0.1.22

### Installation method

cargo install

### Bug description

when using `rumdl check --fix`, `--fail-on any` does not seem to "include" warnings.

### Steps to reproduce

Example document:
```
# Test

[Broken Link](./totally_bugus)
```

rumdl check works correctly:
```
$ rumdl check
test1.md:3:15: [MD057] Relative link './totally_bugus' does not exist

Issues: Found 1 issues in 1 file (18ms)
$ echo $?
1
```

rumdl check --fix shows the problem:
```
$ rumdl check --fix
test1.md:3:15: [MD057] Relative link './totally_bugus' does not exist

Success: No issues found in 1 file (17ms)
$ echo $?
0
```

problem appears to be related to fail-on "any" default. 
```
$ rumdl check --fix --fail-on warning
test1.md:3:15: [MD057] Relative link './totally_bugus' does not exist

Success: No issues found in 1 file (14ms)
$ echo $?
1
```

### Expected behavior

Exit code 1. It would also be nice to avoid the confusing "Success"

### Actual behavior

Exit code 0

### Configuration

```toml

```

### Operating system

macOS

**Comment by rmuir** [comment]
the exit code problem gets cleared up with something like this, but maybe there is a better fix for these booleans:
```diff
--- a/src/commands/check.rs
+++ b/src/commands/check.rs
@@ -163,7 +163,7 @@ pub fn run_check(args: &CheckArgs, global_config_path: Option<&str>, isolated: b
         FailOn::Never => false,
         FailOn::Error => has_errors,
         FailOn::Warning => has_warnings,
-        FailOn::Any => has_issues,
+        FailOn::Any => has_issues || has_warnings || has_errors,
     };

     if should_fail && args.fix_mode != FixMode::Format {
```

**Comment by rvben** [comment]
Fixed in v0.1.23. Rules that depend on ctx.source_file (MD057, MD074) were receiving None during the re-lint phase in fix mode, causing them to silently skip validation. This produced a false "Success: No issues found" with exit code 0 even when violations remained. The source file path is now correctly forwarded through the re-lint.

Thanks @rmuir for raising this!
